cloud.foundry.cli/src/main/java/cloud/foundry/cli/crosscutting/mapping/YamlMapper.java

```diff
@@ -0,0 +1,143 @@
+
+import cloud.foundry.cli.crosscutting.exceptions.RefResolvingException;
+import cloud.foundry.cli.crosscutting.mapping.beans.ApplicationBean;
+import cloud.foundry.cli.crosscutting.mapping.beans.ApplicationManifestBean;
+import cloud.foundry.cli.crosscutting.mapping.beans.Bean;
+import cloud.foundry.cli.crosscutting.mapping.beans.ConfigBean;
+import cloud.foundry.cli.crosscutting.mapping.beans.ServiceBean;
+import cloud.foundry.cli.crosscutting.mapping.beans.SpaceDevelopersBean;
+import cloud.foundry.cli.crosscutting.mapping.beans.SpecBean;
+import cloud.foundry.cli.crosscutting.mapping.beans.TargetBean;
+import org.yaml.snakeyaml.DumperOptions;
+import org.yaml.snakeyaml.Yaml;
+import org.yaml.snakeyaml.constructor.ConstructorException;
+import org.yaml.snakeyaml.constructor.SafeConstructor;
+import org.yaml.snakeyaml.introspector.Property;
+import org.yaml.snakeyaml.nodes.NodeTuple;
+import org.yaml.snakeyaml.nodes.Tag;
+import org.yaml.snakeyaml.representer.Representer;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * This class takes care about loading {@link Bean beans} from configuration files and dumping {@link Bean beans} as
+ * {@link String strings}. It guarantees consistent loading and dumping.
+ */
+public class YamlMapper {
+
+    /**
+     * The indentation (number of white spaces) used for yaml dumping.
+     */
+    public static final int INDENTATION = 2;
+
+    /**
+     * Loads a configuration file as a bean. During this process, the ref-occurrences in the specified configuration
+     * file are resolved.
+     *
+     * @param configFilePath the path to the config file
+     * @param beanType the desired type of the bean to load
+     * @throws IOException if the config file cannot be accessed
+     * @throws RefResolvingException if an error during the ref-resolution process occurs
+     * @throws ConstructorException if the resolved Object can not be dumped as the given Bean type
+     */
+    public static <B extends Bean> B loadBean(String configFilePath, Class<B> beanType) throws IOException {
+        Object yamlTreeRoot = loadYamlTree(configFilePath);
+        yamlTreeRoot = RefResolver.resolveRefs(yamlTreeRoot);
+
+        Yaml treeDumper = createMinimalDumper();
+        String resolvedConfig = treeDumper.dump(yamlTreeRoot);
+
+        Yaml yamlProcessor = new Yaml();
+        return yamlProcessor.loadAs(resolvedConfig, beanType);
+    }
+
+    /**
+     * Reads the file (possibly on a server) and interprets its content as a yaml tree.
+     *
+     * @param filePath the path or url to a file
+     * @return the resulting yaml tree
+     * @throws IOException if the file cannot be accessed
+     */
+    static Object loadYamlTree(String filePath) throws IOException {
+        InputStream inputStream = FileUtils.openLocalOrRemoteFile(filePath);
+        return createTreeLoader().load(inputStream);
```

Comment:
This code should be surrounded with a try with resources block to ensure that the input stream gets reliably closed.